From 065e54f8c59b403ac178fcfe251da8e47aa05b1b Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@lucaweiss.eu>
Date: Sat, 29 Aug 2026 08:20:23 +0200
Subject: [PATCH] Revert "wrappedwindowmanagementpolicy: force inputmethod on
 maliit(6)-server surfaces"

This doesn't compile due to misplaced brackets.

This reverts commit e621dc9c1d8d6ed1eb3bc6e6040b83bc8ea66d7e.
---
 .../mirserver/wrappedwindowmanagementpolicy.cpp    | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp b/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
index 7869594b..9f58d9bc 100644
--- a/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
+++ b/src/platforms/mirserver/wrappedwindowmanagementpolicy.cpp
@@ -120,9 +120,8 @@ namespace qtmir
         int surfaceType = requestParameters.type().is_set() ? requestParameters.type().value() : -1;
         pid_t appPid = miral::pid_of(appInfo.application());
         auto parameters = MinimalWindowManager::place_new_window(appInfo, requestParameters);
-        auto surfaceName = requestParameters.name().is_set() ? requestParameters.name().value() : "";
 
-        if (surfaceType == mir_window_type_inputmethod || surfaceName == "maliit-server" || surfaceName == "maliit6-server") {
+        if (surfaceType == mir_window_type_inputmethod) {
             QFileInfo info(QStringLiteral("/proc/%1/exe").arg(appPid));
 
             // TODO: this is a brute-force fix to make sure only maliit-server can be an input method
@@ -133,11 +132,6 @@ namespace qtmir
                 surfaceType = mir_window_type_normal;
                 parameters.type() = mir_window_type_normal;
             }
-            // TODO: brute-force fix part 2 - force inputmethod for these surfaces since Mir 2.x only sets these on surfaces created with input-method-v1/v2
-            } else {
-                surfaceType = mir_window_type_inputmethod;
-                parameters.type() = mir_window_type_inputmethod;
-            }
         }
 
         bool hasInitialSize = false;
@@ -148,6 +142,12 @@ namespace qtmir
                 parameters.size() = toMirSize(initialSize);
                 hasInitialSize = true;
             }
+
+            auto surfaceName = requestParameters.name().is_set() ? requestParameters.name().value() : "";
+
+            if (surfaceName == "maliit-server") {
+                parameters.type() = mir_window_type_inputmethod;
+            }
         }
 
         if (!hasInitialSize &&
-- 
2.55.0

